File: BEmuJava/bemu/src/main/java/com/bloomberglp/blpapi/MarketMessageServiceStatus.java
//------------------------------------------------------------------------------
// <copyright project="BEmu_maven" file="/BEmu_maven/bemu/src/main/java/com/bloomberglp/blpapi/MarketMessageServiceStatus.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bloomberglp.blpapi;

import com.bloomberglp.blpapi.Message;
import com.bloomberglp.blpapi.CorrelationID;
import com.bloomberglp.blpapi.Name;
import com.bloomberglp.blpapi.Element;

public class MarketMessageServiceStatus extends Message
{
    private final MarketElementString _serviceName;

    MarketMessageServiceStatus(CorrelationID corr) throws Exception
    {
    	super(new Name("ServiceOpened"), corr, null);
    	
        this._serviceName = new MarketElementString("serviceName", "//blp/mktdata");
    }
    
	public int numElements()
	{
		return 1;
	}
    
	public String topicName()
	{
		return "";
	}
	
	public boolean hasElement(String name, boolean excludeNullElements) throws Exception
	{
		return this._serviceName.name().toString().equals(name);
	}
	
	public Element getElement(String name) throws Exception
	{
		if(name.equals(this._serviceName.name().toString()))
			return this._serviceName;
		
		else
			return super.getElement(name);
	}
	
    public String toString()
    {
        StringBuilder result = new StringBuilder();
        result.append(String.format("%s = {%s", this.messageType().toString(), System.getProperty("line.separator")));
        try
        {
			result.append(this._serviceName.prettyPrint(1));
		}
        catch (Exception e) { }
        result.append("}" + System.getProperty("line.separator"));

        return result.toString();
    }

}

File: BEmu/MarketDataRequest/MarketMessageServiceStatus.cs
//------------------------------------------------------------------------------
// <copyright project="BEmu_csh" file="MarketDataRequest/MarketMessageServiceStatus.cs" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

namespace Bloomberglp.Blpapi.MarketDataRequest
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    internal class MarketMessageServiceStatus : Message
    {
        private readonly MarketElementString _serviceName;

        internal MarketMessageServiceStatus(CorrelationID corr) : base(new Name("ServiceOpened"), corr, null)
        {
            this._serviceName = new MarketElementString("serviceName", "//blp/mktdata");
        }

        public override int NumElements { get { return 1; } }
        public override string TopicName { get { return ""; } }
        public override IEnumerable<Element> Elements { get { yield return this._serviceName; } }
        public override Element AsElement { get { return new MarketElementServiceStatus(this); } }
        
        public override object this[string name, int index]
        {
            get
            {
                if (name == this._serviceName.Name.ToString())
                {
                    if (index == 0)
                        return this._serviceName.GetValue();
                }
                return base[name, index];
            }
        }

        public override string ToString()
        {
            StringBuilder result = new StringBuilder();
            result.AppendFormat("{0} = {{{1}", this.MessageType.ToString(), Environment.NewLine);
            result.Append(this._serviceName.PrettyPrint(1));
            result.Append("}" + Environment.NewLine);
            return result.ToString();
        }

    }
}
